### PR TITLE
fixed bug in polyline shorten and extend not reseting the lines

### PR DESCRIPTION
**Pull request by nmaslarinos**
Fixed a bug where calling polyline shorten or extend would alter the polyline points, but not reset the lines attribute. This creates problems where the polylines lines are wrongly displayed and the length attribute returns the previous, now wrong, value. I updated the test documents to also include a check for the length of the resulting polyline.

### What type of change is this?

- [x] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [x] Other (e.g. doc update, configuration, etc)

### Checklist

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas/geometry/curves/polyline.py [hunk 1]
```diff
@@ -578,10 +578,10 @@ def extend(self, length):
         """
         try:
             start, end = length
-            self.points[0] = self.points[0] + self.lines[0].vector.unitized().scaled(-start)
+            self[0] = self[0] + self.lines[0].vector.unitized().scaled(-start)
```
Comment (tomvanmele) [inline src/compas/geometry/curves/polyline.py]: this necessary for the fix?

File: src/compas/geometry/curves/polyline.py [hunk 2]
```diff
@@ -578,10 +578,10 @@ def extend(self, length):
...
         except TypeError:
             start = end = length
-        self.points[-1] = self.points[-1] + self.lines[-1].vector.unitized().scaled(end)
+        self[-1] = self[-1] + self.lines[-1].vector.unitized().scaled(end)
```
Comment (tomvanmele) [inline src/compas/geometry/curves/polyline.py]: same here. if it is not necessary, i find the code more readable with the explicit reference to points...

**Comment by nmaslarinos** [comment]
You are right for both, this is one way of emptying the `polyline._lines` without explicitly doing it. Of course, it can remain as is and simply add a `self._lines = None` before the end of the function

**Review by tomvanmele — APPROVED** [review]

**Comment by gonzalocasas** [comment]
@tomvanmele two questions: 1) ready to merge, right? 2) should I backport this to LTS?

**Comment by tomvanmele** [comment]
yes and yes :)

**Pull request merged**